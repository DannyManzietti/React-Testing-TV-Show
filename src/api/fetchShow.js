import axios from 'axios'

// fetchShow.js
export const fetchShow = () => {
  return axios.get("https://api.tvmaze.com/singlesearch/shows?q=stranger-things&embed=episodes")
     // or res.data, however you want to set that up
     .then(res =>{
     return res
 }).catch(err=> {
 	return err
 })
}
  
 // App.js
// useEffect(() => {
//   fetchShow
//     .then(res => {
//       setSeasons(formatSeasons(res.data._embedded.episodes));
//     }
// }, []);