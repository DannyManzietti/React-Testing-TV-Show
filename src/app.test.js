import React from 'react';
import { render, wait } from '@testing-library/react';
import App from './App'
import userEvent from '@testing-library/user-event'
import {fetchShow as mockFetchShow} from './api/fetchShow'


test('Is the App rendering without crashing', ()=>{
	const mockApp = jest.fn();
	const { getByText, getAllByText, debug , container} = render(<App />);

	expect(container.firstChild).toMatchSnapshot()

	
	getByText(/Fetching data.../i);

}) 




test('User is able to select a season', async() => {
// mockFetchShow.mockResolvedValueOnce(showData);
const { getByText,getAllByText } = render(<App />);

await wait(() =>{
    getByText(/select a season/i)
}) 


userEvent.click(getByText(/select a season/i))
expect(getAllByText(/season /i)).toHaveLength(4);

userEvent.click(getByText(/select a season/i))


expect(getByText(/select a season/i)).toBeTruthy();


userEvent.click(getByText(/select a season/i))
expect(getAllByText(/season /i)).not.toBe( Array);//hmmmmmmm ....huh


});