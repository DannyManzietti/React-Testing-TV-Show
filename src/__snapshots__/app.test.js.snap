// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is the App rendering without crashing 1`] = `
<h2>
  Fetching data...
</h2>
`;
